src/print_graph.cc

```diff
@@ -37,120 +35,136 @@ namespace detail {
 	}
 
 	std::string print_task_graph(const task_ring_buffer& tdag) {
-		std::ostringstream ss;
-		ss << "digraph G { label=\"Task Graph\" ";
+		std::string dot = "digraph G {label=\"Task Graph\" ";
+		auto dot_append = std::back_inserter(dot);
 
 		for(auto tsk : tdag) {
-			std::unordered_map<std::string, std::string> props;
-			props["label"] = "\"" + get_task_label(tsk) + "\"";
-
-			ss << tsk->get_id();
-			ss << "[";
-			for(const auto& it : props) {
-				ss << " " << it.first << "=" << it.second;
-			}
-			ss << "];";
-
+			fmt::format_to(dot_append, "{}[label=\"{}\"];", tsk->get_id(), get_task_label(tsk));
 			for(auto d : tsk->get_dependencies()) {
-				ss << fmt::format("{} -> {} [{}];", d.node->get_id(), tsk->get_id(), dependency_style(d));
+				fmt::format_to(dot_append, "{}->{}[{}];", d.node->get_id(), tsk->get_id(), dependency_style(d));
```

Comment:
Intentional output formatting change?
(Probably not too relevant, reading the dot files manually is not a common use case)

Reply:
The dot file is not for human consumption, so there's no reason to keep this whitespace.